README.md

```diff
@@ -100,15 +105,39 @@ If loading an existing Autobase then set `bootstrap` to `base.key`, otherwise pa
 
 ```js
 {
-  open: store => { ... }, // create the view
-  apply: (nodes, view, hostcalls) => { ... }, // handle nodes
-  close: view => { ... }, // close the view
+  open: (store, host) => { ... }, // create the view
+  apply: async (nodes, view, host) => { ... }, // handle nodes to update view
+  optimistic: false, // Autobase supports optimistic appends
+  close: async view => { ... }, // close the view
   valueEncoding, // encoding
   ackInterval: 1000 // enable auto acking with the interval
+  encryptionKey: buffer, // Key to encrypt the base
+  encrypt: false, // Encrypt the base if unencrypted & no encryptionKey is set
+  encrypted: false, // Expect the base to be encrypted, will throw an error otherwise, defaults to true if encryptionKey is set
+  fastForward: true, // Enable fast forwarding. If passing { key: base.core.key }, they autobase will fastforward to that key first.
+  wakeup: new ProtomuxWakeup(), // Set a custom wakeup protocol for hinting which writers are active, see `protomux-wakeup` for protocol details
 }
 ```
 
-An `ackInterval` may be set to enable automatic acknowledgements. When enabled, in cases where it would help the linearizer converge the base shall eagerly append `null` values to the oplog.
+An `ackInterval` may be set to enable automatic acknowledgements. When enabled, in cases where it would help the linearizer converge, the base shall eagerly append `null` values to merge causal forks.
+
+Setting an autobase to be `optimistic` means that writers can append an `optimistic` block even when they are not a writer. For a block to be optimistically applied to the view, the writer must be acknowledge via `host.ackWriter(key)`.
```

Comment:
Note here that you should only use this to implement self verifiable blocks, ie this block can be verified against the base by proving a signature or similar, failure to do this is catastrophic and means anyone in the world can append to your autobase (if using optimistic blocks)

Reply:
@mafintosh to clarify, the self verifying mechanism should be at the application logic level, correct? Ie. Autobase doesn't need a particular verification internally to process the block. The verification is to prevent random writers changing the view via optimistic blocks (when optimistic appends are supported) by blocking unverifiable blocks in the `apply` function?

Reply:
Using keet as an example, verification would maybe include a way to verify that the peer has a valid invite but hasn't been added as a writer yet?

Reply:
correct, its a "if you know what you are doing v powerful, if not prop dont use it" feature

Reply:
@mafintosh Added f36dc16 to clarify & updated the example to reflect the warning